aws/structure.go

```diff
@@ -1026,6 +1026,15 @@ func expandFloat64Map(m map[string]interface{}) map[string]*float64 {
 	return float64Map
 }
 
+// Expands a map of string to interface to a map of string to *string
+func expandStringMap(m map[string]interface{}) map[string]*string {
```

Comment:
Nit: It's confusingly named but [documented](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/data-handling-and-conversion.md#root-typemap-of-typestring-and-aws-map-of-string), this is equivalent to `stringMapToPointers()` -- probably worth migrating everything to this name in a separate PR.

Reply:
ahah i should of referred to the docs..i looked and looked 😅 . will do then 👍